flink-learning-common/src/test/java/com/zhisheng/common/utils/DateUtilTests.java

```diff
@@ -0,0 +1,87 @@
+package com.zhisheng.common.utils;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateUtilTests {
+
+    @Test
+    public void testFormat() {
+        Date date1 = new Date(1556665200000L);
+        Date date2 = new Date(1556723220000L);
+        Date date3 = new Date(1556723235000L);
+        Date date4 = new Date(1556723235000L);
+
+        Assert.assertEquals(date1.getTime(), DateUtil.format(date1));
+
+        Assert.assertEquals("2019-05-01", DateUtil.format(date1.getTime(), DateUtil.YYYY_MM_DD));
+        Assert.assertEquals("2019-05-01", DateUtil.format(date1, DateTimeZone.getDefault(), DateUtil.YYYY_MM_DD));
+
+        Assert.assertEquals(date1.getTime(), DateUtil.format("2019-05-01", DateUtil.YYYY_MM_DD));
+        Assert.assertEquals(date2.getTime(), DateUtil.format("2019-05-01 16:07", DateUtil.YYYY_MM_DD_HH_MM));
+        Assert.assertEquals(date3.getTime(), DateUtil.format("2019-05-01 16:07:15", DateUtil.YYYY_MM_DD_HH_MM_SS));
+        Assert.assertEquals(date4.getTime(), DateUtil.format("2019-05-01 16:07:15.0", DateUtil.YYYY_MM_DD_HH_MM_SS_0));
+    }
+
+    @Test
+    public void testIsValidDate() {
+        Assert.assertTrue(DateUtil.isValidDate("2019-05-01", DateUtil.YYYY_MM_DD));
+
+        Assert.assertFalse(DateUtil.isValidDate("01-05-2019", DateUtil.YYYY_MM_DD));
+    }
+
+    @Test
+    public void testToDate() {
+        Assert.assertEquals(new Date(1556665200000L),
+                DateUtil.toDate("2019-05-01", DateUtil.YYYY_MM_DD));
+    }
+
+    @Test
+    public void testWithTimeAtStartOfDay() {
+        DateTimeFormatter dtf = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date date = new Date(1557874800000L);
+        DateTime dt = new DateTime(1557874800000L);
+
+        Assert.assertEquals(sdf.format(date), DateUtil.withTimeAtStartOfDay(date, dtf));
+        Assert.assertEquals(sdf.format(date), DateUtil.withTimeAtStartOfDay(dt, dtf));
+    }
+
+    @Test
+    public void testWithTimeAtEndOfDay() {
+        DateTimeFormatter dtf = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date date = new Date(1557961199000L);
+        DateTime dt = new DateTime(1557961199000L);
+
+        Assert.assertEquals(sdf.format(date), DateUtil.withTimeAtEndOfDay(date, dtf));
```

Comment:
Can you just put the string in the expected? instead of calling `sdf.format`

Reply:
Done